src/javascript/binary/base/page.js

```diff
@@ -468,12 +470,58 @@ Header.prototype = {
 
         this.menu.register_dynamic_links();
     },
+    start_clock_ws : function(){
+        var that = this;
+        var clock_handle;
+        var query_start_time;
+        var clock = $('#gmt-clock');
+
+        function init(){
+            BinarySocket.send({ "time": 1});
+            query_start_time = (new Date().getTime());  
+        }
+        BinarySocket.init({
+            onmessage : function(msg){
+                var response = JSON.parse(msg.data);
+                if (response && response.msg_type === 'time') {
+                    responseMsg(response);
```

Comment:
You can put function's body inside this condition and not to use function responseMsg at all. There is no point to initiate the function if you call it only once